dev/benchmarking/ddl/1_tables.sql

```diff
@@ -0,0 +1,291 @@
+  , machine_name citext NOT NULL UNIQUE
+  , memory_bytes bigint NOT NULL
+  , cpu_actual_frequency_Hz bigint NOT NULL
+  , cpu_id integer NOT NULL
+  , gpu_id integer NOT NULL
+  , os_id integer NOT NULL
+  , FOREIGN KEY (cpu_id) REFERENCES public.cpu
+  , FOREIGN KEY (gpu_id) REFERENCES public.gpu
+  , FOREIGN KEY (os_id) REFERENCES public.os
+);
+COMMENT ON TABLE public.machine
+  IS 'Unique identifiers for a machine.';
+COMMENT ON COLUMN public.machine.machine_name
+  IS 'A unique machine name of your choice.';
+COMMENT ON COLUMN public.machine.memory_bytes
+  IS 'The output of `sysctl -n hw.memsize`.';
+COMMENT ON COLUMN public.machine.cpu_actual_frequency_Hz
+  IS 'The output of `sysctl -n hw.cpufrequency`.';
+
+-- UNIT
+CREATE TABLE IF NOT EXISTS public.unit
+(
+  id SERIAL PRIMARY KEY
+  , unit_name citext NOT NULL UNIQUE
+  , benchmark_type_id integer NOT NULL
+  , FOREIGN KEY (benchmark_type_id) REFERENCES public.benchmark_type(id)
+);
+COMMENT ON TABLE public.unit IS 'The actual units for a reported benchmark.';
+COMMENT ON COLUMN public.unit.unit_name
+  IS 'For example: ms, ns, microseconds.';
+
+-- ENVIRONMENT
+CREATE TABLE IF NOT EXISTS public.environment
+(
+  id SERIAL
+  , benchmark_language_id integer NOT NULL
+  , language_version_id integer NOT NULL
+  , language_dependencies_id integer NOT NULL
+  , PRIMARY KEY
+      (id, benchmark_language_id, language_version_id)
+  , FOREIGN KEY
+      (benchmark_language_id)
+      REFERENCES public.benchmark_language
+  , FOREIGN KEY
+      (language_version_id, benchmark_language_id)
+      REFERENCES public.language_version(id, benchmark_language_id)
+  , FOREIGN KEY
+      (language_dependencies_id, benchmark_language_id)
+      REFERENCES public.language_dependencies(id, benchmark_language_id)
+);
+COMMENT ON TABLE public.environment
+  IS 'Identifies a build environment for a specific suite of benchmarks.';
+
+-- BENCHMARK
+CREATE TABLE IF NOT EXISTS public.benchmark
+(
+  id SERIAL
+  , benchmark_language_id integer NOT NULL
+  , benchmark_name citext NOT NULL
+  , parameter_names text[]
+  , lessisbetter boolean NOT NULL
+  , benchmark_description text NOT NULL
+  , benchmark_version citext NOT NULL
+  , unit_id integer NOT NULL
+  , PRIMARY KEY (id, benchmark_language_id)
+  , FOREIGN KEY (benchmark_language_id) REFERENCES public.benchmark_language
+  , FOREIGN KEY (unit_id) REFERENCES public.unit
+);
+COMMENT ON TABLE public.benchmark
+  IS 'Identifies an individual benchmark.';
+COMMENT ON COLUMN public.benchmark.parameter_names
+  IS 'A list of strings identifying the parameter names in the benchmark.';
+COMMENT ON COLUMN public.benchmark.lessisbetter
+  IS 'True if a smaller benchmark value is better.';
+COMMENT ON COLUMN public.benchmark.benchmark_version
+  IS 'Can be any string. In Airspeed Velocity, the version is '
+     'by default the hash of the entire code string for the benchmark.';
+
+-- BENCHMARK_RESULT
+CREATE TABLE IF NOT EXISTS public.benchmark_result
```

Comment:
bikeshedding: what do you think of "run" or "execution" instead of name for result

Reply:
Not bikeshedding! Stuff like this is a couple of lines now, but hours of headache downstream. I am super-thankful you are willing to take the time to make this better even in its minutiae.